backend/app/Http/Controllers/MarketplaceController.php

```diff
@@ -82,6 +82,28 @@ public function putTransaction($card_id)
         return $this->getCardDetail($card_id);
     }
 
+    public function sellCards()
+    {
+        //add selling car to listing
+        $user = auth()->user();
+        $data = json_decode(Request::getContent(), true);
+        foreach ($data as $cardId) {
```

Comment:
Do not assume that the user passes a valid set of card IDs.

We need to verify that:
- The card exists
- The card is available for auction (not already on auction, not in battle, etc)
- The card is owned by the user trying to sell the card (thus the `$listing->user_id should not be set to the $user->id`